src/Utilities/hooks/useTableTools/__tests__/reportParser.test.js

```diff
@@ -0,0 +1,69 @@
+import cloneDeep from 'lodash/cloneDeep';
+import { parseJobReport, prepareItems } from '../reportParser';
+import {
+  fixturesExtendedReport,
+  fixturesPlainReport,
+} from './__fixtures__/jobResultsItems.fixtures';
+
+describe('parseJobReport', () => {
+  it('should parse a job without report', () => {
+    const parsed = parseJobReport(fixturesPlainReport);
+
+    expect(parsed).toHaveLength(1);
+    expect(parsed).toMatchSnapshot();
+  });
+
+  it('should parse a job with report_json', () => {
+    const parsed = parseJobReport(fixturesExtendedReport);
+
+    expect(parsed).toHaveLength(8);
+    parsed.forEach(({ issue_parsed }) => {
+      expect(issue_parsed).toMatchSnapshot();
```

Comment:
What do you mean by being "ignored"?

Reply:
I mean, when the snapshot changes are huge it will be harder to understand the difference between the old change and the new change. As the snapshot testing is done inside a loop, I expect there will be many of them

Reply:
@mkholjuraev, If there are changes in the snapshots, then we should review them and see if it is intended (as with any other snapshots). I put this assertion to generate the expected data easier. Not sure I see any issue with that.

We could perhaps replace it with assertions against the fixtures: so that the output values are correctly parsed and correspond to input data.

Reply:
I also find it a bit add to take the snapshot within a loop, when it could simply be:

```
  expect(parsed).toMatchSnapshot();
```

Reply:
@mkholjuraev, @bastilian, in the last commits, I have replaced it with 
```
expect(parsed.map(({ issue_parsed }) => issue_parsed)).toMatchSnapshot();
```
I am not sure we can use more specific assertions here since it's the purpose of the parser function to compose simpler data structure from initial fixtures (that have complex structure, see extraction logic for remediations or issue diagnosis). We just need to simply verify that in the output we get the expected values. Since there are 8 of different issues, I don't see a reason to write separate assertions for each field. If the snapshot changes anyhow, this is a responsibility for a dev to understand why it has happened, and I think test description/assertions are descriptive enough.